loki/bulk/scheduler.py

```diff
@@ -629,7 +629,8 @@ def process(self, transformation, reverse=False, item_filter=SubroutineItem, use
             if use_file_graph:
                 for node in traversal:
                     items = graph.nodes[node]['items']
-                    transformation.apply(items[0].source, item=items[0], items=items)
+                    transformation.apply(items[0].source, item=items[0], items=items,
+                                         targets=items[0].targets)
```

Comment:
I use `targets` to filter callstatements on line 117. Yes I see what you are saying. This will break for example if there are multiple subroutines in the same file. How about this:
```
for node in traversal:
    for item in graph.nodes[node]['items']:
        if item_filter and not isinstance(item, item_filter):
            continue
        transformation.apply(item.source, item=item, targets=item.targets)
```